x-pack/solutions/security/plugins/security_solution/public/siem_migrations/rules/components/rules_table_columns/actions.tsx

```diff
@@ -34,33 +36,43 @@ const ActionName = ({
   openMigrationRuleDetails,
   installMigrationRule,
 }: ActionNameProps) => {
-  const { navigateToApp } = useKibana().services.application;
-  if (migrationRule.elastic_rule?.id) {
-    const ruleId = migrationRule.elastic_rule.id;
-    return (
-      <EuiLink
-        disabled={disableActions}
-        onClick={() => {
-          navigateToApp(APP_UI_ID, {
-            deepLinkId: SecurityPageName.rules,
-            path: getRuleDetailsUrl(ruleId),
-          });
-        }}
-        data-test-subj="viewRule"
-      >
-        {i18n.ACTIONS_VIEW_LABEL}
-      </EuiLink>
-    );
-  }
+  const { getUrlForApp, navigateToUrl } = useKibana().services.application;
+
+  const hrefRuleDetails = useMemo(
+    () =>
+      getUrlForApp(APP_UI_ID, {
+        deepLinkId: SecurityPageName.rules,
+        path: getRuleDetailsUrl(migrationRule.elastic_rule?.id ?? ''),
+      }),
+    [getUrlForApp, migrationRule.elastic_rule?.id]
+  );
 
+  const goToRuleDetails = useCallback(
+    (event: React.MouseEvent) => {
+      if (!isModifiedEvent(event) && isLeftClickEvent(event)) {
+        event.preventDefault();
+        navigateToUrl(hrefRuleDetails);
+      }
+    },
+    [hrefRuleDetails, navigateToUrl]
+  );
+
+  // Failed
   if (migrationRule.status === SiemMigrationStatus.FAILED) {
+    return null;
+  }
+
+  // Installed
+  if (migrationRule.elastic_rule?.id) {
     return (
-      <EuiLink disabled={disableActions} onClick={() => {}} data-test-subj="restartRule">
-        {i18n.ACTIONS_RESTART_LABEL}
+      // eslint-disable-next-line @elastic/eui/href-or-on-click
+      <EuiLink href={hrefRuleDetails} onClick={goToRuleDetails} data-test-subj="viewRule">
+        {i18n.ACTIONS_VIEW_LABEL}
```

Comment:
NIT:
```suggestion
      <SecuritySolutionLinkAnchor deepLinkId={SecurityPageName.rules} path={getRuleDetailsUrl(migrationRule.elastic_rule?.id ?? '')} data-test-subj="viewRule">
        {i18n.ACTIONS_VIEW_LABEL}
```
And we can remove all the link related callbacks and logic

Reply:
done in https://github.com/elastic/kibana/pull/206285/commits/a63cecc37daa67d9915b00df3ac8606769e6f1d0